### PR TITLE
[#23] As a user, I can access TIL app via nimblehq.co/til

### PR DESCRIPTION
**Pull request by hoangmirs**
Closes https://github.com/nimblehq/til-web/issues/23

## What happened 👀

Configure til-web as the subdirectory of `nimblehq.co` and can access it via `nimblehq.co/til`

## Insight 📝

- First, we need to configure the [basepath](https://nextjs.org/docs/api-reference/next.config.js/basepath) for til-web. We can access the application with `/til` path
- Add some configurations to load images properly. I have to put the assets inside `public/til` folder as I tried many ways but it did not work.
- Then using [proxies feature](https://docs.netlify.com/routing/redirects/rewrites-proxies/) from Netlify to access `til` via `nimblehq.co/til`. It supports proxy Netlify applications

## Proof Of Work 📹

You can access til-web application on nimblehq-web staging:
https://62a1a5b36a58c93d312670d1--staging-nimblehq.netlify.app/til

### CODE REVIEW

**Review by Lahphim — APPROVED** [review]
It seems that the preview doesn't work 🤨

**Comment by hoangmirs** [comment]
> It seems that the preview doesn't work 🤨

You need to access it via https://deploy-preview-26--nimble-til.netlify.app/til as we're using basepath in this PR

**Comment by hoangmirs** [comment]
> * The images seem not to load locally on my machine.
> * For a better development experience I think we can redirect `localhost:3000`to `localhost:3000/till`.  For this, we can use Next's [redirect with basepath support](https://nextjs.org/docs/api-reference/next.config.js/rewrites#rewrites-with-basepath-support) so that we don't get any `404` on '/`:
> 
> ```js
> # next.config.js
>   async redirects() {
>     return process.env.NODE_ENV === 'development'
>       ? [
>           {
>             source: '/',
>             destination: '/til',
>             basePath: false,
>             permanent: false,
>           },
>         ]
>       : [];
>   },
> ```
> 
> I think it'll be nice to have all the routes i.e `localhost:3000/page/2` redirect to `localhost:3000/til/page/2` locally too. What do you think? @hoangmirs

Thanks for your suggestion 👏 
I added it in https://github.com/nimblehq/til-web/pull/26/commits/28ff7692995801417e807c10b4159d5fbfaa2552
I think we don't need to check the environment as it just runs on development.

**Comment by Lahphim** [comment]
> > * The images seem not to load locally on my machine.
> > * For a better development experience I think we can redirect `localhost:3000`to `localhost:3000/till`.  For this, we can use Next's [redirect with basepath support](https://nextjs.org/docs/api-reference/next.config.js/rewrites#rewrites-with-basepath-support) so that we don't get any `404` on '/`:
> > 
> > ```js
> > # next.config.js
> >   async redirects() {
> >     return process.env.NODE_ENV === 'development'
> >       ? [
> >           {
> >             source: '/',
> >             destination: '/til',
> >             basePath: false,
> >             permanent: false,
> >           },
> >         ]
> >       : [];
> >   },
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I think it'll be nice to have all the routes i.e `localhost:3000/page/2` redirect to `localhost:3000/til/page/2` locally too. What do you think? @hoangmirs
> 
> Thanks for your suggestion 👏 I added it in [28ff769](https://github.com/nimblehq/til-web/commit/28ff7692995801417e807c10b4159d5fbfaa2552) I think we don't need to check the environment as it just runs on development.

@hoangmirs Do they any helpers managing with the basepath for applying to the markdown file?
It would be something like this here.
```
...
coverImage: '{__basePath__}/assets/posts/example/cover.png'
...
```

**Comment by hoangmirs** [comment]
> @hoangmirs Do they any helpers managing with the basepath for applying to the markdown file? It would be something like this here.
> 
> ```
> ...
> coverImage: '{__basePath__}/assets/posts/example/cover.png'
> ...
> ```

@Lahphim We haven't had it yet. But as we're putting all assets in `public/til` folder, then I think we can keep the current paths. If change to like that, we need to update the assets files as well 🤔

File: src/components/Image/index.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import NextImage, { ImageProps } from 'next/image';
+
+const Image = ({ src, children, ...rest }: ImageProps) => {
+  const isProd = process.env.NODE_ENV === 'production';
+  const finalSrc = isProd ? src : `${process.env.NEXT_PUBLIC_BASE_PATH}${src}`;
+
+  return (
+    <NextImage src={finalSrc} data-test-id="image-wrapper" {...rest}>
```
Comment (tyrro) [inline src/components/Image/index.tsx:8]: Maybe we could spare the responsibility of defining the `data-test-id` to the calling component? The reason is, that makes testing the image `src` a lot easier:

```javascript
    const props = {
      src: '/image.svg',
      alt: 'image',
      'data-test-id': 'image',
    };

    render(<Image {...props} />);

    const image = screen.getByTestId(props['data-test-id']);

    expect(image).toBeVisible();
    expect(image).toHaveAttribute('src', props.src);
```

File: src/components/Image/index.test.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+import Image from '.';
+
+describe('Image', () => {
+  it('renders the image properly', () => {
+    const { container } = render(
+      <Image src="https://picsum.photos/200/300" alt="test" layout="fill" />
+    );
+
+    expect(container).toMatchSnapshot();
```
Comment (tyrro) [inline src/components/Image/index.test.tsx:11]: If we agree on the below comment, then we can test the image `src` easily:

```javascript
const src = "https://picsum.photos/200/300"

expect(container).toBeVisible();
expect(container).toHaveAttribute('src', props.src);
```

**Review by tyrro — APPROVED** [review]

**Pull request merged**